model-optimizer/mo/utils/find_ie_version.py

```diff
@@ -0,0 +1,134 @@
+"""
+
+import os
+import sys
+import platform
+import subprocess
+
+lib_env_key = "PATH" if platform.system() == "Windows" else "LD_LIBRARY_PATH"
+if lib_env_key not in os.environ:
+    os.environ[lib_env_key] = ""
+
+python_path_key = "PYTHONPATH"
+if python_path_key not in os.environ:
+    os.environ[python_path_key] = ""
+
+lib_path_orig = os.environ[lib_env_key]
+python_path_orig = os.environ[python_path_key]
+
+
+def setup_env(module="", libs=[]):
+    """
+    Update os.environ variables with given values.
+    :param module: path to python module
+    :param libs: list with paths to libraries
+    """
+    os.environ[python_path_key] = os.pathsep.join([os.environ[python_path_key], module])
+    os.environ[lib_env_key] = os.pathsep.join([os.environ[lib_env_key], *libs])
+
+
+def reset_env():
+    """
+    Reset os.environ variables to default values
+    """
+    os.environ[python_path_key] = python_path_orig
+    os.environ[lib_env_key] = lib_path_orig
+
+
+def try_to_import_ie(module="", libs=[]):
+    """
+    Check if IE python modules exists and in case of success
+    environment will be set with given values.
+    :param module: path to python module
+    :param libs: list with paths to libraries
+    """
+    path_to_script = os.path.join(os.path.realpath(os.path.dirname(__file__)), 'check_ie_bindings.py')
+    # We need to execute python modules checker in subprocess to avoid issue with environment
+    # in case if previous import was unsuccessful it can fail further imports even if sys.path
+    # will be restored to initial default values.
+    # To pass environment to sub-process PATH/LD_LIBRARY_PATH and PYTHONPATH are used from
+    # os.environ that is set after setup_env()
+    setup_env(module=module, libs=libs)
+    status = subprocess.run([sys.executable, path_to_script], env=os.environ)
+    if status.returncode == 0:
+        return True
+    else:
+        reset_env()
+        return False
+
+
+def find_ie_version():
+    """
+    Tries to import IE python bindings. In case of successful import
+    PATH/LD_LIBRARY_PATH and PYTHONPATH environment variables will be set
+    This variables must be passed to subprocess in order to execute IE python bindings.
+    Example:
+        if find_ie_version():
+            subprocess.run([sys.executable, path_to_script], env=os.environ)
+
+    """
+    print("[ IMPORT ] Checking existing InferenceEngine python API")
+    if try_to_import_ie():
+        return True
+
+    python_version = 'python{}.{}'.format(sys.version_info[0], sys.version_info[1])
+
+    script_path = os.path.realpath(os.path.dirname(__file__))
+    bindings_paths = [
+        # Windows
+        {
+            "module": os.path.join(script_path, '../../../../python/', python_version),
+            "libs": [
+                os.path.join(script_path, '../../../inference_engine/bin/intel64/Release'),
+                os.path.join(script_path, '../../../inference_engine/external/tbb/bin'),
```

Comment:
But if we built IE from source, we have another paths. Maybe my previous comment is not relevant.

Reply:
We don't need path to libraries if we build from sources. So we have just path to python modules.